src/saveload/industry_sl.cpp

```diff
@@ -26,18 +26,28 @@ static const SaveLoad _industry_desc[] = {
 	    SLE_VAR(Industry, location.h,                 SLE_FILE_U8 | SLE_VAR_U16),
 	    SLE_REF(Industry, town,                       REF_TOWN),
 	SLE_CONDNULL( 2, 0, 60),       ///< used to be industry's produced_cargo
-	SLE_CONDARR(Industry, produced_cargo,             SLE_UINT8,  2,              78, SL_MAX_VERSION),
-	SLE_CONDARR(Industry, incoming_cargo_waiting,     SLE_UINT16, 3,              70, SL_MAX_VERSION),
-	    SLE_ARR(Industry, produced_cargo_waiting,     SLE_UINT16, 2),
-	    SLE_ARR(Industry, production_rate,            SLE_UINT8,  2),
+	SLE_CONDARR(Industry, produced_cargo,             SLE_UINT8,   2,              78, 199),
+	SLE_CONDARR(Industry, produced_cargo,             SLE_UINT8,  16,             200, SL_MAX_VERSION),
+	SLE_CONDARR(Industry, incoming_cargo_waiting,     SLE_UINT16,  3,              70, 199),
+	SLE_CONDARR(Industry, incoming_cargo_waiting,     SLE_UINT16, 16,             200, SL_MAX_VERSION),
+	SLE_CONDARR(Industry, produced_cargo_waiting,     SLE_UINT16,  2,               0, 199),
```

Comment:
I think this is correct, but not entirely sure, converting SLE_ARR to SLE_CONDARR starting at version 0.

Reply:
I think the savegames stores the size of the array.
So it may not even need the condition. Just changing 2 to 16 may be enough.
But needs trying and possibly proper initialisation of the array with CT_INVALID.